ddtrace/internal/_encoding.pyx

```diff
@@ -41,20 +47,31 @@ class BufferItemTooLarge(Exception):
     pass
 
 
-cdef inline int array_prefix_size(int l):
+cdef inline const char * string_to_buff(str s):
```

Comment:
a comment here might be nice, since this helper will "borrow" the `char*` used by the `PyObject`, so if the `PyObject` gets GC'd then this `char*` will become invalid.

It should be fine for this specific case